tests/publiccloud/patch_and_reboot.pm

```diff
@@ -23,17 +23,19 @@ sub run {
 
     my $cmd_time = time();
     my $ref_timeout = check_var('PUBLIC_CLOUD_PROVIDER', 'AZURE') ? 3600 : 240;
+    my $remote = $args->{my_instance}->username . '@' . $args->{my_instance}->public_ip;
+
     # pkcon not present on SLE-micro
     kill_packagekit($args->{my_instance});
     $args->{my_instance}->ssh_script_retry("sudo zypper -n --gpg-auto-import-keys ref", timeout => $ref_timeout, retry => 6, delay => 60);
     record_info('zypper ref time', 'The command zypper -n ref took ' . (time() - $cmd_time) . ' seconds.');
     record_soft_failure('bsc#1195382 - Considerable decrease of zypper performance and increase of registration times') if ((time() - $cmd_time) > 240);
-
-    fully_update_system($args->{my_instance});
+    ssh_fully_patch_system($remote) or fully_update_system($args->{my_instance});
```

Comment:
```suggestion
    if(is_sle_micro()) {
           fully_update_system($args->{my_instance});
     } else {
           ssh_fully_patch_system($remote);
      }
```

This "or" is very confusing when you read the code , much more clear is like this . than you don't need to go to function to figure out which one will be executed

Reply:
This OR is a simple command output check, also used in bash.
I preferred this form also to keeping the code more similar to previous existing code avoiding enveloping routine call in a sub-level of if, The `condition check` is performed inside the routine also as resilience to unwanted usage, similar to other existing routines.

Reply:
well maybe code looks better like this , but it make it less readable . `ssh_fully_patch_system($remote) or fully_update_system($args->{my_instance});` you have 0 chances to understand which one will be actually executed and what does it mean . So when I will read this code to find root cause of some problem I will be forced to find implementation of both functions and read what actually happening there to understand which one will be executed .

Reply:
ok, I will change it